dali/sasha/saxref.cpp

```diff
@@ -670,6 +678,60 @@ class CNewXRefManagerBase
 
     }
 
+    void startHeartbeat(const char * op)
+    {
+        CriticalBlock block(heartbeatSect);
+        time(&heartbeatStartTime);
+        lastHeartbeat = heartbeatStartTime;
+        heartbeatInterval = 60; // Start with 1 minute (in seconds)
+        heartbeatCount = 0;
+        uncondlog("%s heartbeat started (interval: 1 minute)", op);
+    }
+
+    void checkHeartbeat(const char * op)
+    {
+        time_t now = time(NULL);
+        bool logHeartbeat = false;
+
+        {
+            CriticalBlock block(heartbeatSect);
+            if ((now - lastHeartbeat) >= heartbeatInterval)
+            {
+                heartbeatCount++;
+                lastHeartbeat = now;
+                logHeartbeat = true;
+            }
+        }
+
+        if (logHeartbeat)
+        {
+            unsigned elapsedMinutes = (unsigned)(now - heartbeatStartTime) / 60;
+            unsigned elapsedHours = elapsedMinutes / 60;
+            unsigned remainingMinutes = elapsedMinutes % 60;
+
+            // Format UTC timestamp directly from time_t
+            struct tm *utc_tm = gmtime(&now);
+            char timestamp[32];
+            strftime(timestamp, sizeof(timestamp), "%Y-%m-%d %H:%M:%S", utc_tm);
+
+            if (elapsedHours > 0)
+                uncondlog("%s - elapsed: %uh %um (%s UTC)", op, elapsedHours, remainingMinutes, timestamp);
+            else
+                uncondlog("%s - elapsed: %um (%s UTC)", op, elapsedMinutes, timestamp);
+
+            // Increase heartbeat interval with cap of 1 hour
+            if (heartbeatCount < 7)
```

Comment:
would be clearer to do away with this count, and rely on the current intreval, e.g.
```
unsigned currentIntervalMs = heartbeatTimer.queryPeriodMs();
if (currentPeriodMs < 3600000)
{
    constexpr unsigned oneHourMs = 3600000;
    unsigned newIntervalMs = currentIntervalMs >= oneHourMs / 2 ? oneHourMs : currentIntervalMs * 2;
```